client/src/components/all-components/CodeBlock/StyleComponents/HTMLTag.jsx

```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { htmlTagRegex, htmlElementRegex } from "../Constants";
+
+/**
+ * React component for rendering HTML-like code with syntax highlighting for HTML tags and attributes.
+ *
+ * The `HTMLTag` component takes a single prop, `children`, which should be a string containing
+ * HTML-like code to be displayed with syntax highlighting. It applies syntax highlighting to the text,
+ * distinguishing between HTML tags and attributes to make the code more readable and understandable.
+ *
+ * @param {Object} props - The props for the `HTMLTag` component.
+ * @param {string} props.children - The React element representing the content of the code line, in this
+ * case, the children will simply be passed in text, which will then be styled and syntactically highlighted similarly to HTML-like code.
+ *
+ * @return {JSX.Element} The rendered component that displays HTML-like code with syntax highlighting.
+ */
+
+const HTMLTag = ({ children }) => {
+  if (typeof children === "object") {
```

Comment:
@dm9965

Reply:
If you pass in a prop as part of the displayed string, it is passed in as an object and would break the application otherwise, so you have to create a workaround for it.